plugin/hosts/hostsfile.go

```diff
@@ -106,7 +133,7 @@ func (h *Hostsfile) readHosts() {
 		return
 	}
 
-	newMap := h.parse(file, h.inline)
+	newMap := h.parse(file, h.hmap, h.options.autoReverse)
```

Comment:
just give parse the whole options struct

Reply:
elsewhere this function is used to merge the inline data with the file and the code is
```h.hmap = h.parse(r, inline, h.options.autoReverse)```
so changing the API would not allow this anymore AFAICS.

Reply:
I would just make that change, there is no backwards consideration for code living inside a plugin.

Reply:
Ok, in that case, I think the code works only with the first parameter (removing the second and third). I first added a log.Fatalf in the `if override == nil {` before any change and none of the test code triggered it. It probably need a good pair of eyes to check this again. I left the change as a separate patch for ease of review.